### ISSUE TITLE
Allow custom route names

### ISSUE THREAD

**Issue by mwpastore** [mwpastore/sinja#8]
Enhance the `resource`, `has_one`, and `has_many` keywords to take an options hash, which may include a custom route name to be used when drawing namespaces.

**Comment by jgnagy** [comment]
This may be the solution to my problem, but is there no way to stop sinja from deciding where to put dashes in controller names (routes)? I have updated my test repo (which happily works now thanks to #12 being resolved), but it takes the `TestKey` and forces the route to be `/test-key`. What if I didn't want a dash, and instead simply wanted `/testkey`? Is this configurable? If not, maybe this is something I could help with / submit a PR for.

**Comment by mwpastore** [comment]
It draws the routes based on the symbol passed to the `resource` keyword. The symbol is dasherized and pluralized. Currently, there's no way to override this behavior, which is what this issue is intended to track. It's something I'm willing to prioritize if it helps an early adopter!

All that being said, your resource is named `:keys`, so the resource route should be `/keys`, not `/test-keys`. I think you may be looking at the links in the response payload. These are actually generated by JSONAPI::Serializers, and it **is** customizable! Just override the `type` method in your serializer class:

```ruby
class TestKeySerializer
  include JSONAPI::Serializer

  attribute :name
  attribute :created_at

  def type
    'testkey'
  end
end
```

More information is available [here](https://github.com/fotinakis/jsonapi-serializers#more-customizations).

**Comment by jgnagy** [comment]
Yes, that helped quite a bit for test scenario (I committed my update to reflect that), but unfortunately I think it needs to be done at a deeper level in sinja because it is failing in my application while including a related `has_many` resource. In my real use-case, I have `GET /users/1?include=apikeys%2Cidentities` (bringing along both `apikeys` and `identities`), but the response payload doesn't seem to rely on the underlying APIKey's `type` instance method when creating the `has_many` collection.

I am trying to build an app that relies on your library to implement the {json:api} spec, so I don't know if that makes me an early adopter, but I would certainly appreciate this customization being available. I can successfully POST to /apikeys and create a new key, but subsequently listing a user's keys produces unexpected results. Going from dashes to underscores in a consistent manner across libraries and possible languages is difficult, so being able to explicitly dictate the route would be helpful.

**Comment by mwpastore** [comment]
Nothing you've describe makes me think you need this feature. Do you have some time to chat with me on [Gitter](https://gitter.im/sinja-rb/Lobby) or somewhere else so I can help you work through this use case?
